llama-index-integrations/vector_stores/llama-index-vector-stores-redis/llama_index/vector_stores/redis/base.py

```diff
@@ -191,29 +182,136 @@ def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        if field.type not in REDIS_LLAMA_FIELD_SPEC:
+            raise ValueError(f"Unsupported field type {field.type} for {field.name}")
+
+        field_info = REDIS_LLAMA_FIELD_SPEC[field.type]
+
+        # Check for unsupported operator
+        if filter.operator not in field_info["operators"]:
+            raise ValueError(
+                f"Filter operator {filter.operator} not supported for {field.name} of type {field.type}"
+            )
+
+        # Create field instance and apply the operator function
+        field_instance = field_info["class"](field.name)
+        return field_info["operators"][filter.operator](field_instance, filter.value)
+
+    def _create_redis_filter_expression(
+        self, metadata_filters: MetadataFilters
+    ) -> FilterExpression:
+        """
+        Generate a Redis Filter Expression as a combination of metadata filters.
+
+        Args:
+            metadata_filters (MetadataFilters): List of metadata filters to use.
+
+        Returns:
+            FilterExpression: A Redis filter expression.
+        """
+        filter_expression = FilterExpression("*")
+        if metadata_filters.filters:
+            for filter in metadata_filters.filters:
+                # Index must be created with the metadata field in the index schema
+                field = self._index.schema.fields.get(filter.key)
+                if not field:
+                    logger.warning(
+                        f"{filter.key} field was not included as part of the index schema, and thus cannot be used as a filter condition."
+                    )
+                    continue
+                # Extract redis filter
+                redis_filter = self._to_redis_filter(field, filter)
+                # Combine with conditional
+                if metadata_filters.condition == "and":
+                    filter_expression = filter_expression & redis_filter
+                else:
+                    filter_expression = filter_expression | redis_filter
+        return filter_expression
+
+    def _to_redis_query(self, query: VectorStoreQuery) -> VectorQuery:
+        """Creates a RedisQuery from a VectorStoreQuery."""
+        filter_expression = self._create_redis_filter_expression(query.filters)
+        return VectorQuery(
+            vector=query.query_embedding,
+            vector_field_name=self._vector_field_name,
+            return_fields=self._return_fields,
+            num_results=query.similarity_top_k,
+            filter_expression=filter_expression,
+        )
+
+    def _extract_node_and_score(self, doc, redis_query: VectorQuery):
+        """Extracts a node and its score from a document."""
+        try:
+            # TODO investigate this a bit more
+            node = metadata_dict_to_node({"_node_content": doc["_node_content"]})
+            node.text = doc[self._text_field_name]
+        except Exception:
+            # Handle legacy metadata format
+            node = TextNode(
```

Comment:
tbh, I think we just need get the text here -- no need to worry about the rest (this relates to handling arbitrary indexes built by redis)

Reply:
Got it, can you clarify what this would need to look like? Thanks!!